app/src/main/java/com/github/se/project/model/lesson/LessonViewModel.kt

```diff
@@ -31,49 +36,55 @@ class LessonsViewModel(private val repository: LessonRepository) : ViewModel() {
         object : ViewModelProvider.Factory {
           @Suppress("UNCHECKED_CAST")
           override fun <T : ViewModel> create(modelClass: Class<T>): T {
-            return LessonsViewModel(LessonRepositoryFirestore(Firebase.firestore)) as T
+            return LessonViewModel(LessonRepositoryFirestore(Firebase.firestore)) as T
           }
         }
   }
 
   /**
-   * Retrieves all lessons associated with a specific user.
+   * Generates a new unique ID.
    *
-   * @param userUid The UID of the user whose lessons are to be retrieved.
+   * @return A new unique ID.
    */
-  fun getLessonsByUser(userUid: String) {
-    repository.getLessonsByUserId(
-        userUid,
-        onSuccess = { lessons_.value = it },
-        onFailure = { e -> Log.e("LessonViewModel", "Error loading user: $userUid's lessons", e) })
+  fun getNewUid(): String {
+    return repository.getNewUid()
   }
 
   /**
    * Adds a new lesson to the repository.
    *
    * @param userUid The UID of the user to associate with the lesson.
    * @param lesson The Lesson object to be added.
+   * @param onComplete Callback to execute when the operation completes.
    */
-  fun addLesson(userUid: String, lesson: Lesson) {
-    repository.addLessonByUserId(
-        userUid = userUid,
+  fun addLesson(userUid: String, lesson: Lesson, onComplete: () -> Unit) {
+    repository.addLesson(
         lesson = lesson,
-        onSuccess = { getLessonsByUser(userUid) }, // Refresh the lesson list on success
-        onFailure = { Log.e("LessonViewModel", "Error adding lesson: $lesson", it) })
+        onSuccess = {
+          onComplete() // Call the provided callback on success
```

Comment:
We should maybe also update the mutable list currentUserLesson if the lesson was added successfully so that the other classes that uses the field (like Overviewscreen I think) will be updated consequently. 
This can be done by calling getLessonsForTutor of getLessonsForStudent (but this require knowing if the current user is a tutor or a student), or by simply append the new Lesson to the list (if we assume the current user is the one who added the lesson, which if I am not mistaken we can assume without problem)

Reply:
This is true, but the idea was to add a callback function to update (call one of the getLesson) the lesson list via this callback. Since we have 2 different getLessons (one for tutor and one for student), we don't know which one to call and this is why the idea was to call this via the callback function.

Reply:
Okay I see but I think it is not really secure because it will be easy to forget that onComplete should call the corresponding getLessons when we use these methods. Maybe another suggestion would be to replace the onComplete function by a Boolean, for example isTutor, so then we can call the corresponding getLessonsMethod and be sure that it is called when we add/delete/update a lesson. I think it may be more sure and less prone to inattention errors later on.